src/client/core/page-unload-barrier.js

```diff
@@ -80,6 +83,20 @@ function prolongUnloadWaiting (timeout) {
     }, timeout);
 }
 
+function waitForFailDownload () {
+    return new Promise(resolve => {
+        nativeMethods.setTimeout.call(window, () => {
+            // NOTE: we use flag to confirm file downloading because if unload raised browser can respond empty string
+            transport
+                .queuedAsyncServiceMsg({ cmd: MESSAGE.waitForFileDownload })
+                .then(isFileDownloaded => {
```

Comment:
I think we should rename this because it's in fact this flag doesn't mean `file is downloading`. It may be `downloadingStarted` or `fileDonwloadTriggered (handled)` or something else..